tests/dashboard/plot_orchestrator_test.py

```diff
@@ -1126,5 +1189,534 @@ def test_add_cell_immediately_remove_it_then_workflow_commits_no_error(
         plot_orchestrator.remove_plot(cell_id)
 
         # Workflow commits
-        job_number = uuid.uuid4()
-        fake_job_orchestrator.simulate_workflow_commit(workflow_id, job_number)
+        commit_workflow_for_test(job_orchestrator, workflow_id, workflow_spec)
+
+
+class TestLateSubscriberPlotRetrieval:
+    """Test that late subscribers can retrieve existing plots via get_cell_state."""
+
+    def test_get_cell_state_returns_none_for_cell_without_plot(
+        self, plot_orchestrator, plot_cell
+    ):
+        """get_cell_state should return (None, None) for cell without plot."""
+        grid_id = plot_orchestrator.add_grid(title='Test Grid', nrows=3, ncols=3)
+        cell_id = plot_orchestrator.add_plot(grid_id, plot_cell)
+
+        # Cell exists but workflow hasn't committed yet
+        plot, error = plot_orchestrator.get_cell_state(cell_id)
+        assert plot is None
+        assert error is None
+
+    def test_get_cell_state_returns_plot_after_workflow_commits(
+        self,
+        plot_orchestrator,
+        plot_cell,
+        workflow_id,
+        workflow_spec,
+        job_orchestrator,
+        fake_data_service,
+    ):
+        """get_cell_state should return plot after workflow commits."""
+        grid_id = plot_orchestrator.add_grid(title='Test Grid', nrows=3, ncols=3)
+        cell_id = plot_orchestrator.add_plot(grid_id, plot_cell)
+
+        # Commit workflow
+        job_ids = commit_workflow_for_test(job_orchestrator, workflow_id, workflow_spec)
+        job_number = job_ids[0].job_number
+
+        # Simulate data arrival for ALL sources
+        import scipp as sc
+
+        from ess.livedata.config.workflow_spec import JobId, ResultKey
+
+        for source_name in plot_cell.config.source_names:
+            result_key = ResultKey(
+                workflow_id=workflow_id,
+                job_id=JobId(source_name=source_name, job_number=job_number),
+                output_name=plot_cell.config.output_name,
+            )
+            fake_data_service[result_key] = sc.scalar(1.0)
+
+        # Now get_cell_state should return the plot
+        plot, error = plot_orchestrator.get_cell_state(cell_id)
+        assert plot is not None
+        assert isinstance(plot, FakePlot)
+        assert error is None
+
+    def test_get_cell_state_returns_error_when_plot_creation_fails(
+        self,
+        workflow_id,
+        workflow_spec,
+        job_orchestrator,
+        fake_data_service,
+        fake_stream_manager,
+    ):
+        """get_cell_state should return error when plot creation fails."""
+        # Create plotting controller configured to fail
+        failing_controller = FakePlottingController(stream_manager=fake_stream_manager)
+        failing_controller.configure_to_raise(ValueError("Plot creation failed"))
```

Comment:
Is it worth parametrizing the test with different kinds of errors (RuntimeError, ValueError, etc.), or does it not matter?
(I was thinking in case some try/except block somewhere misses one of the errors fx)